microsoft/adaptivecards-templates
# Users/hagupt/designer save

This PR is for the save functionality of a template.

Work done:
- created store.currentTemplate actions/reducer in redux store to handle a specific template in focus on preview page (in the future) and designer page
- "New Template" button dispatches redux action to handle creation of new template
- Created "Save" and "Publish" button UI inside of the designer component
- "Save" button successfully posts template to the backend using client SDK from a redux-thunk. If a template is being saved for the first time, the template is assigned an ID by the backend, which is reflected in the redux store and the designer props. if a template has been saved before, the redux thunk action ensures that pre-existing template on the backend is updated rather than a new template being created.

To test:
1. Go to Dashboard first
2. Hit "New Template"
3. Edit template on designer
4. Hit save on the designer
5. Check Postman to confirm the template was saved (get all templates and then it should be at the bottom in the response).
6. Note the ID of the posted template.
7. Modify template again in the designer and save.
8. Confirm on Postman again that the template updated *with the same ID as before*.
9. Go to Dashboard and confirm that you see the template you created (it should be at the bottom).
